src/main/java/teammates/ui/template/CourseEditInstructorPanel.java

```diff
@@ -0,0 +1,210 @@
+package teammates.ui.template;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import teammates.common.datatransfer.InstructorAttributes;
+import teammates.common.util.Const;
+
+public class CourseEditInstructorPanel {
+    private int index;
+    private InstructorAttributes instructor;
+    private List<ElementTag> permissionInputGroup1;
+    private List<ElementTag> permissionInputGroup2;
+    private List<ElementTag> permissionInputGroup3;
+    private ElementTag resendInviteButton;
+    private ElementTag editButton;
+    private ElementTag deleteButton;
+    private ElementTag addSectionLevelForInstructorButton;
+    private boolean isAccessControlDisplayed;
+    private List<CourseEditSectionRow> sectionRows;
+    
+    public CourseEditInstructorPanel(int instructorToShowIndex, int instructorIndex, 
+                                     InstructorAttributes instructor, InstructorAttributes currentInstructor,
+                                     List<String> sectionNames, List<String> feedbackNames) {
+        if (instructorToShowIndex == -1) { 
+            index = instructorIndex;
+            isAccessControlDisplayed = false;
+        } else {
+            index = instructorToShowIndex;
+            isAccessControlDisplayed = true;
+        }
+        this.instructor = instructor;
+        
+        createSectionRows(instructorIndex, sectionNames, feedbackNames);
+        permissionInputGroup1 = createPermissionInputGroup1ForInstructorPanel(instructor);
+        permissionInputGroup2 = createPermissionInputGroup2ForInstructorPanel(instructor);
+        permissionInputGroup3 = createPermissionInputGroup3ForInstructorPanel(instructor);
```

Comment:
Keep it consistent whether to create the class-scope object in the method, or to create a method-scope object and pass it back.